include/boost/dynamic_bitset/dynamic_bitset.hpp

```diff
@@ -1070,7 +1070,7 @@ bool dynamic_bitset<Block, Allocator>::all() const
                 return false;
             }
         }
-        block_type const mask = ~(~static_cast<Block>(0) << extra_bits);
+        block_type const mask = ~(static_cast<Block>(~0) << extra_bits);
```

Comment:
So much strangeness here.  You're making a value, casting it to a `Block`, and then assigning it to a `block_type`. If they were different types, you'd be doing a conversion. But they're not - so why use two different names?
Also, we know that `block_type` is an unsigned integral type, right? So why start with a signed constant `0`?
Why not just write:
     `block_type const mask = (block_type(1) << extra_bits) - 1;`
Isn't that the same value? No more `~`s, no `static_cast`.